server/server.go

```diff
@@ -279,19 +297,38 @@ func (s *Server) concurrentRequest(ctx context.Context, conf map[string]lookout.
 			cs, err := send(ctx, a.Client, settings)
 			if err != nil {
 				aLogger.Errorf(err, "analysis failed")
+				if s.ExitOnError {
+					errCh <- err
+				}
 				return
 			}
 
 			if len(cs) == 0 {
 				aLogger.Infof("no comments were produced")
+				return
 			}
 
-			comments.Add(a.Config, cs...)
+			result = &lookout.AnalyzerComments{
+				Config:   a.Config,
+				Comments: cs,
+			}
 		}(name, a)
 	}
-	wg.Wait()
 
-	return comments.Get()
+	var comments []lookout.AnalyzerComments
+	for i := 0; i < len(s.analyzers); i++ {
+		select {
+		case err := <-errCh:
+			cancel()
```

Comment:
Is this needed if we already do the `defer` above?

Reply:
somehow in my head cancellation was meant to be before return from this function. But there is no reason for that. Thanks. Removed.